src/utils/streams/__tests__/intersperse_stream.js

```diff
@@ -0,0 +1,38 @@
+import expect from 'expect.js';
+import sinon from 'sinon';
+
+import {
+  createPromiseFromStreams,
+  createListStream,
+  createIntersperseStream,
+  createConcatStream
+} from '../';
+
+describe('intersperseStream', () => {
+  it('places the intersperse value between each provided value', async () => {
+    expect(
+      await createPromiseFromStreams([
+        createListStream(['to', 'be', 'or', 'not', 'to', 'be']),
+        createIntersperseStream(' '),
+        createConcatStream()
+      ])
+    ).to.be('to be or not to be');
+  });
+
+  it('emits the intersperse value right before the second value, does not wait', async () => {
```

Comment:
I was trying to ensure that this isn't holding onto chunks it doesn't need to. I'll pick a better title